command/sub8_missions/missions/start_gate.py

```diff
@@ -21,96 +21,98 @@
 
 from sub8 import sub_singleton
 
-FOUND_START_GATE = False
-SPEED = 0.3
 
 fprint = sub8_tools.text_effects.FprintFactory(
-    title="START_GATE", msg_color="white").fprint
+title="START_GATE", msg_color="white").fprint
 
-@txros.util.cancellableInlineCallbacks
-def run(sub_singleton):
-    fprint('Starting mission')
-    fprint('Assuming camera is facing gate')
-    fprint('Leveling off', msg_color="yellow")
-    yield sub_singleton.move.zero_roll_and_pitch().go()
-
-    start_gate_enable = yield sub_singleton.nh.get_service_client('/vision/start_gate/enable', SetBool)
-    fprint('Turning on vision service')
-    yield start_gate_enable(SetBoolRequest(data=True))
-
-    start_gate_search = yield sub_singleton.nh.get_service_client('/vision/start_gate/pose', VisionRequest2D)
-    fprint("Searching for start gate pose")
-    start_gate_search_res = yield start_gate_search(VisionRequest2DRequest(target_name=''))
-
-    if not start_gate_search_res.found:
-        fprint("Waiting a few seconds and trying again:")
-        time.sleep(3)
-        start_gate_search_res = yield start_gate_search(VisionRequest2DRequest(target_name=''))
-    # yield start_gate_enable(SetBoolRequest(data=False))
-    # yield start_gate_enable(SetBoolRequest(data=True))
-    if not start_gate_search_res.found:
-        fprint("Running search pattern")
-        global FOUND_START_GATE
-        while not FOUND_START_GATE and start_gate_search_res.pose.x == 0:
-            fprint(FOUND_START_GATE)
-            time.sleep(1)
-            fprint("AGAIN - Searching for start gate pose")
-            start_gate_find(sub_singleton)
-            yield search_pattern(sub_singleton)
-    
-    fprint("Found start gate: " + str(start_gate_search_res.pose))
-    yield align_for_dummies(sub_singleton, start_gate_search_res)
-    fprint("YOLO")
-
-    while(FOUND_START_GATE):
-        start_gate_find(sub_singleton)
-        yield sub_singleton.move.forward(.3).zero_roll_and_pitch().go(speed=SPEED)
-
-    fprint("Finished")
+class StartGateMission:
 
+    FOUND_START_GATE = False
```

Comment:
`FOUND_START_GATE` isn't actually a constant.

Can you make speed a ros param?